planex/cmd/pin.py

```diff
@@ -139,15 +147,32 @@ def main(argv=None):
     spec = load_spec_and_lnk(xs_path, package_name)
     pin = get_pin_content(args, spec)
 
-    if not args.quiet:
+    if not (args.quiet or args.show or args.unpin):
         print(json.dumps(pin, indent=2, sort_keys=True))
 
+    default_output = "PINS/{}.pin".format(package_name)
     output = args.output
     if args.write:
-        output = "PINS/{}.pin".format(package_name)
+        output = default_output
 
     if output is not None:
         path = os.path.dirname(output)
         makedirs(path)
         with open(output, "w") as out:
             json.dump(pin, out, indent=2, sort_keys=True)
+
+    if args.show:
+        try:
+            with open(default_output, 'r') as f:
+                print(f.read())
+        except IOError as e:
+            if e.errno not in (errno.ENOENT,):
+                raise
+            print("Package '{}' is not pinned".format(package_name))
+            pass
+
+    if args.unpin:
+        try:
+            os.remove(default_output)
+        except:
+            pass
```

Comment:
I think it would be better to fail ungracefully if the removal goes wrong

Reply:
For any other reason than ENOENT, maybe. Unpinning a thing which is already not pinned should be a no-op.

Reply:
Why not test for the existence of the file first? This isn't a command where we need to worry much about concurrency.

Reply:
It was easier to just catch the error, given that I have some code doing just that right above it to borrow.